training/model_management/src/azureml/model/mgmt/downloader/downloader.py

```diff
@@ -44,18 +43,36 @@ def __init__(self, model_id: str):
         """
         self._model_id = model_id
         self._model_uri = self.HF_ENDPOINT + f"/{model_id}"
-        self._model_info = None
         self._hf_api = HfApi(endpoint=self.HF_ENDPOINT)
+        self._is_valid_id = None
+        self._model_info = None
+
+    @property
+    def is_valid_id(self):
+        """Returns true if Hugging face model id is valid."""
+        if self._is_valid_id is None:
+            try:
+                model_list = [
+                    model.modelId for model in self._hf_api.list_models(filter=ModelFilter(model_name=self._model_id))
+                ]
+                self._is_valid_id = self._model_id in model_list
+            except Exception as e:
+                raise ValueError(f"Failed to validate model id : {e}")
+
+        return self._is_valid_id
 
     @property
     def model_info(self) -> ModelInfo:
         """Hugging face model info."""
         if not self._model_info:
-            model_list: List[ModelInfo] = self._hf_api.list_models(filter=ModelFilter(model_name=self._model_id))
-            for info in model_list:
-                if self._model_id == info.modelId:
-                    self._model_info = info
-                    break
+            try:
```

Comment:
we are doing the same operation twice. May be we can store model info in the first try and use it?